packages/spacecat-shared-gpt-client/src/clients/genvar-client.js

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { createUrl } from '@adobe/fetch';
+import { ImsClient } from '@adobe/spacecat-shared-ims-client';
+import { hasText, isObject, isValidUrl } from '@adobe/spacecat-shared-utils';
+
+import { fetch as httpFetch } from '../utils.js';
```

Comment:
please use `tracingFetch` from `spacecat-shared-utils` - the `../utils.js` can probably be eliminated, adjust other usage in this package.